yesod-core/src/Yesod/Core/Content.hs

```diff
@@ -226,6 +240,53 @@ typeOctet = "application/octet-stream"
 simpleContentType :: ContentType -> ContentType
 simpleContentType = fst . B.break (== _semicolon)
 
+decoderForCharset :: Maybe B.ByteString -> L.ByteString -> TL.Text
+decoderForCharset (Just encodingSymbol)
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "utf-8")        = LE.decodeUtf8With EE.lenientDecode
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "US-ASCII")     = TL.fromStrict . fst . decodeASCIIPrefix . B.toStrict
+  | encodingSymbol == (encodeUtf8 $ T.pack $ "latin1")       = LE.decodeLatin1
```

Comment:
i generally do not recommend using a formatter that aligns `=` like that - very sensitive to alignment breaking in other cases

instead, indenting after the `=` helps to have alignment on the expressions

```suggestion
  | encodingSymbol == (encodeUtf8 $ T.pack $ "latin1") = 
      LE.decodeLatin1
```

this is one of those things where the very nice aesthetics of Mathy Lookin Haskell Code don't play super nice with code-on-computer (vs code-on-paper)